lisa/tests/base.py

```diff
@@ -835,19 +837,74 @@ def get_trace(self, **kwargs):
         """
         return Trace(self.trace_path, self.plat_info, **kwargs)
 
-class RTATestBundle(FtraceTestBundle):
+class DmesgTestBundle(TestBundle):
     """
-    Abstract Base Class for :class:`lisa.wlgen.rta.RTA`-powered TestBundles
-
-    .. seealso: :class:`lisa.tests.base.FtraceTestBundleMeta` for default
-        ``ftrace_conf`` content.
+    Abstract Base Class for TestBundles based on dmesg output.
     """
-
     DMESG_PATH = 'dmesg.log'
     """
     Path to the dmesg log in the result directory.
     """
 
+    @property
+    def dmesg_path(self):
+        """
+        Path to the dmesg output log file
+        """
+        return ArtifactPath.join(self.res_dir, self.DMESG_PATH)
+
+    @property
+    def dmesg_entries(self):
+        """
+        List of parsed dmesg output entries
+        :class:`devlib.trace.dmesg.KernelLogEntry`.
+        """
+        with open(self.dmesg_path) as f:
+            return [
+                KernelLogEntry.from_str(line)
+                for line in f.read().splitlines()
+                if line.strip()
+            ]
+
+    def test_dmesg(self, level='warn', facility=None) -> ResultBundle:
+        """
+        Basic test on kernel dmesg output.
+
+        :param level: Any dmesg entr with a level more critical than (and
+            including) that will make the test fail.
+        :type level: str
+
+        :param facility: Only select entries emitted by the given dmesg
+            facility like `kern`. Note that not all versions of `dmesg` are
+            able to print it, so specifying it may lead to no entry being
+            inspected at all. If ``None``, the facility is ignored.
+        type facility: str or None
+        """
+        levels = DmesgCollector.LOG_LEVELS
+        # Consider as an issue all levels more critical than `level`
+        issue_levels = levels[:levels.index(level) + 1]
```

Comment:
This only makes sense if message priority decreases with the level, is that the case?

Reply:
Yes, LOG_LEVELS is explicitly defined with the right order:
```
    # taken from "dmesg --help"
    # This list needs to be ordered by priority
    LOG_LEVELS = [
        "emerg",        # system is unusable
        "alert",        # action must be taken immediately
        "crit",         # critical conditions
        "err",          # error conditions
        "warn",         # warning conditions
        "notice",       # normal but significant condition
        "info",         # informational
        "debug",        # debug-level messages
    ]
```